src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideExtensions.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using Acuminator.Utilities.Common;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PXOverrideMismatch
+{
+	public static class PXOverrideExtensions
+	{
+		public static INamedTypeSymbol GetFirstTypeArgument(this INamedTypeSymbol type)
+		{
+			if (type == null)
+			{
+				return null;
+			}
+
+			if (type.TypeArguments.Any())
+			{
+				return type.TypeArguments.FirstOrDefault() as INamedTypeSymbol;
+			}
+
+			return type.ContainingType.GetFirstTypeArgument();
+		}
+
+		public static INamedTypeSymbol GetPXGraphExtension(this INamedTypeSymbol typeSymbol, INamedTypeSymbol pxGraphExtensionType)
```

Comment:
I have several remarks about this helper.
1. We keep public utility methods that could be reused in the Utility project in a suitable namespace and class. 
2. We throw argument null exception if the input is null. No need to support case with null argument.
3. This check spends too much resources. There is no need for recursive check here which will call inherits helper which already checks the whole type hierarchy. 
4. If what you are trying to obtain is PXGraphExtension<TBaseExxtension, ...., , TGraph> then this code is not correct too. It doesn't cover cases like this:
```csharp
public class SomeExtension<TGraph, TDac> : PXGraphExtension<TGraph> {}
```
It is generic and is inherited from `PXGraphExtension`

Overall, I don't see a need in this method if you will redesign analyzer to be the aggregated graph analyzer

Reply:
We use GetGraphExtensionWithBaseExtensions now, these helpers are removed. Thank you!